activerecord/lib/active_record/internal_metadata.rb

```diff
@@ -0,0 +1,56 @@
+require 'active_record/scoping/default'
+require 'active_record/scoping/named'
+
+module ActiveRecord
+  # This class is used to create a table that keeps track of values and keys such
+  # as which environment migrations were run in.
+  class InternalMetadata < ActiveRecord::Base
+    class << self
+      def primary_key
+        "key"
+      end
+
+      def table_name
+        "#{table_name_prefix}#{ActiveRecord::Base.internal_metadata_table_name}#{table_name_suffix}"
+      end
+
+      def index_name
+        "#{table_name_prefix}unique_#{ActiveRecord::Base.internal_metadata_table_name}#{table_name_suffix}"
+      end
+
+      def store(hash)
+        hash.each do |key, value|
+          first_or_initialize(key: key.to_s).update_attributes!(value: value)
+        end
+      end
```

Comment:
I have mixed feelings about this. I don't want it to feel like hash access because it is hitting the database. I want to avoid "Oh, look a hash, hashes are fast. We can put lots of stuff in it" reaction. I also don't want people to accidentally try to use multiple level keys etc. It doesn't look like a hash because it isn't one. I can change if you like `[]` better though.